tests/Barcoder.Tests/DataMatrix/DataMatrixEncoderTests.cs

```diff
@@ -180,6 +180,154 @@ public void Encode_Gs1ModeEnabled_ValidGs1Content_NonPreDefinedApplicationIdenti
             }
         }
 
+        [Fact]
+        public void Encode_8x18()
+        {
+            var content = "123456";
+            var expectedDataBits = ImageStringToBools(@"
+                #.#.#.#.#.#.#.#.#.
+                ##..#.....##.....#
+                ##...#..##.####.#.
+                ##..##...#...###.#
+                ####.##..###..#...
+                #.####...#...#.###
+                #....####.##.##.#.
+                ##################
+            ");
+
+            TestEncode_FixedSize(8, 18, content, expectedDataBits);
+        }
+
+        [Fact]
+        public void Encode_8x32()
+        {
+            var content = "Abc12345";
+            var expectedDataBits = ImageStringToBools(@"
+                #.#.#.#.#.#.#.#.#.#.#.#.#.#.#.#.
+                #.##....#..##.###..##..#..###..#
+                ##..#.##.####.#.#.#####..#####..
+                #.###..#.......#####....#.###.##
+                #...#.....#####.#.###..##..###..
+                ####..#..#.#.####.###.####.#...#
+                #.##.#..#..#....##.###.#...##...
+                ################################
+            ");
+
+            TestEncode_FixedSize(8, 32, content, expectedDataBits);
+        }
+
+        [Fact]
+        public void Encode_12x26()
+        {
+            var content = "Abc12345Gtzw";
+            var expectedDataBits = ImageStringToBools(@"
+                #.#.#.#.#.#.#.#.#.#.#.#.#.
+                #.##....##..###.###.###.##
+                ##..#.##....##.#.#.#.##...
+                #.###...#..#...#.#.#.##.##
+                #...#..###.#...#.#.#..#...
+                #.##..#....#..###.#.....##
+                ####.########.#.##.###....
+                #.#.#.###.#.###.#.#...##.#
+                #..##.###.##.###.##..#....
+                #.##.#####.##..#.#..###.##
+                ##..#...#.#...#.#......#..
+                ##########################
+                
+            ");
+
+            TestEncode_FixedSize(12, 26, content, expectedDataBits);
+        }
+
+        [Fact]
+        public void Encode_12x36()
+        {
+            var content = "Abc12345Hij!klm";
+            var expectedDataBits = ImageStringToBools(@"
+                #.#.#.#.#.#.#.#.#.#.#.#.#.#.#.#.#.#.
+                #.##....#.##.##.#######......###....
```

Comment:
^ this test data is incorrect on this line; the very last character should be a "#", and not a ".".

this fixes this unit test. unfortunately, I could not find test data errors in the other codes.